server.js

```diff
@@ -1,33 +1,130 @@
-import express from 'express'
-import bodyParser from 'body-parser'
-import cors from 'cors'
-
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
-
-// Defines the port the app will run on. Defaults to 8080, but can be 
+import express from "express";
+import bodyParser from "body-parser";
+import cors from "cors";
+import data from "./data/netflix-titles.json";
+
+
+// Defines the port the app will run on. Defaults to 8080, but can be
 // overridden when starting the server. For example:
 //
 //   PORT=9000 npm start
-const port = process.env.PORT || 8080
-const app = express()
+const port = process.env.PORT || 8080;
+const app = express();
 
 // Add middlewares to enable cors and json body parsing
-app.use(cors())
-app.use(bodyParser.json())
+app.use(cors());
+app.use(bodyParser.json());
 
 // Start defining your routes here
-app.get('/', (req, res) => {
-  res.send('Hello world')
-})
+app.get("/", (req, res) => {
+  res.send(`Welcome to the Netflix API with ${data.length} listed movies and TV shows.`);
+});
+
+// List both movies and TV shows.
+app.get("/all", (req, res) => {
+  res.json(data);
+});
+
+// List movies
+app.get("/movies", (req, res) => {
+  const movies = data.filter((item) => item.type === "Movie");
+  res.json(movies);
+});
+
+// List TV shows
+app.get("/tv-shows", (req, res) => {
+  const tvShows = data.filter((item) => item.type === "TV Show");
+  res.json(tvShows);
+});
+
+// Return a movie or tv show that matches an id
+app.get("/id/:id", (req, res) => {
+  const { id } = req.params;
+  const itemFound = data.find((item) => item.show_id === +id);
+
+  itemFound
+    ? res.json(itemFound)
+    : res.send(`Sorry, couldn't find a movie or TV show with ID ${id}`);
+});
+
+// List all titles or only the once matching the title keyword
+app.get("/titles", (req, res) => {
+  const { title } = req.query;
+  if (!title) {
+    res.json(data.map((item) => item.title));
+  } else {
+    const titleToLowerCase = title.toLowerCase();
+    const filteredByTitle = data.filter((item) =>
+      item.title.toString().toLowerCase().includes(titleToLowerCase)
```

Comment:
nice that you made your input case-insensitive with making them all lower case 🥇